db/migrate/20220428071505_add_first_reply_activity_at_to_conversation.rb

```diff
@@ -0,0 +1,23 @@
+class AddFirstReplyActivityAtToConversation < ActiveRecord::Migration[6.1]
+  def change
+    add_column :conversations, :first_reply_created_at, :datetime, default: nil
+    add_index :conversations, :first_reply_created_at
+
+    add_first_reply_activity_at_to_conversations
+  end
+
+  private
+
+  def add_first_reply_activity_at_to_conversations
+    ::Conversation.find_in_batches do |conversation_batch|
+      Rails.logger.info "Migrated till #{conversation_batch.first.id}\n"
+      conversation_batch.each do |conversation|
+        # rubocop:disable Rails/SkipsModelValidations
+        if conversation.messages.outgoing.count.positive?
```

Comment:
This logic needs to be updated after the campaign triggers first_response bug fix.
https://github.com/chatwoot/chatwoot/pull/4607

Reply:
We will have to test this backfill migration against prod data and see how much time it will take.